ddtrace/opentracer/util.py

```diff
@@ -5,3 +5,16 @@ def merge_dicts(x, y):
     z = x.copy()   # start with x's keys and values
     z.update(y)    # modifies z with y's keys and values & returns None
     return z
+
+
+def get_reasonable_service_name():
```

Comment:
Let's move these functions into `ddtrace.utils` package because they may be used in general, and are not OpenTracing related. Also, always use a plural for a module containing multiple utilities. I'd say to call this function `get_application_name` because, in the end, we're trying to do that.

Also I think this function may be different. If you call this function multiple times from different part of your stack, I guess you're having wrong/different results. I may suggest (if it makes sense):
* move it in `ddtrace.utils`
* `ddtrace.__init__` is the entrypoint for everything
* here we can call the function once so that it's executed always from the same entrypoint (that is when we import the `ddtrace` module for the first time
* even if it seems that it doesn't make sense, you can store the first call in the `utils.whatever` module you'll use, so that it will give always the right result.

Another approach is to simply expose the attribute, but since it's a function, I would just say to store the result after the first execution done always from the same entrypoint. Of course, this change needs to be tested and see if `ddtrace.__init__.py` is the good candidate.

Makes sense?

Reply:
Okay, so from what I've uncovered so far `ddtrace.__init__.py` is fine for manual usage. Issues arise when tracing via the `ddtrace-run` command. When `ddtrace-run` executes, `ddtrace.__init__.py` is invoked in `sitecustomize.py` which doesn't give us any useful application or file name.

I'm going to continue digging further into this.